packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -304,6 +304,34 @@ export class ConversationService extends TypedEventEmitter<Events> {
     return this.establishMLSGroupConversation(groupId, qualifiedUsers, selfUserId, selfClientId, qualifiedId);
   }
 
+  /**
+   * Centralized handler for scenarios where an MLS conversation is detected as broken.
+   * It resets the conversation and then invokes the provided callback so callers can retry
+   * their original operation (e.g., re-adding/removing users, re-joining, etc.) with the new group id.
+   *
+   * Contract:
+   * - input: conversationId to reset; callback invoked after reset with the new group id
+   * - output: the value returned by the callback
+   * - error: throws if reset fails or new group id is missing
+   */
+  private async handleBrokenMLSConversation<T = void>(
+    conversationId: QualifiedId,
+    afterReset?: (newGroupId: string) => Promise<T>,
+  ): Promise<T> {
+    const {
+      conversation: {group_id: newGroupId},
+    } = await this.resetMLSConversation(conversationId);
+    if (!newGroupId) {
+      const errorMessage = 'Tried to reset MLS conversation but no group_id found in response';
+      this.logger.error(errorMessage, {conversationId});
+      throw new Error(errorMessage);
+    }
+    if (afterReset) {
+      return afterReset(newGroupId);
+    }
+    return undefined as unknown as T;
```

Comment:
we dont do that please. 

'as' is already bad enough :)
"as unknown as" is absolutely not ok.

Reply:
https://github.com/wireapp/wire-web-packages/pull/7405/commits/5e9a3df566ca80116b2930cdb51889a5ce9fdd85